### PR TITLE
perf(ivy): class bindings benchmark

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
This PR adds a new benchmark focused on `class` manipulation + benchpress scripts.

### CODE REVIEW

File: modules/benchmarks/src/class_bindings/app.component.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <button id="create" (click)="create()">Create</button>
+    <button id="update" (click)="update()">Update</button>
+    <button id="destroy" (click)="destroy()">Destroy</button>
+    <class-bindings *ngIf="show" [msg]="msg" [list]="list"><class-bindings>
+  `
+})
+export class AppComponent {
+  show = false;
+  msg = 'hello';
+  list = new Array(1000).fill(0).map((_, i) => ({i, text: 'foobar' + i}));
```
Comment (mhevery) [inline modules/benchmarks/src/class_bindings/app.component.ts]: I believe this creates holley array. See https://github.com/angular/angular/pull/32155

**Review by mhevery — APPROVED** [review]

File: modules/benchmarks/src/class_bindings/app.component.ts (hunk 1 above)
Comment (pkozlowski-opensource) [reply modules/benchmarks/src/class_bindings/app.component.ts]: fixed

File: modules/benchmarks/src/class_bindings/class_bindings.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    <span class="baz">one</span>
+    <span class="qux">two</span>
+    <div>
+      <span class="qux">three</span>
+      <span class="qux">four</span>
+      <span class="baz">five</span>
+      <div>
+        <span class="qux">six</span>
+        <span class="baz">seven</span>
+        <span [class]="msg">eight</span>
```
Comment (IgorMinar) [inline modules/benchmarks/src/class_bindings/class_bindings.component.ts]: It's a bit surprising that a variable called "msg" is bound to class attr. If we change the msg value to contain a space the benchmark results could suddenly be very different.

**Review by IgorMinar — APPROVED** [review]
One nit, but I'm not sure if it's worth changing. Lgtm

File: modules/benchmarks/src/class_bindings/class_bindings.component.ts (hunk 2 above)
Comment (pkozlowski-opensource) [reply modules/benchmarks/src/class_bindings/class_bindings.component.ts]: yeh, I've kept the names from the original benchmark, not sure it worth changing...

**Pull request closed without merging**